src/polyglot-notebooks-vscode-common/src/metadataUtilities.ts

```diff
@@ -301,9 +301,27 @@ export function getRawNotebookDocumentMetadataFromNotebookDocumentMetadata(noteb
         rawMetadata.polyglot_notebook = notebookDocumentMetadata;
     }
 
+    merge(rawMetadata, documentRawMetadata);
     return rawMetadata;
 }
 
+function merge(destination: { [key: string]: any }, source: { [key: string]: any }) {
+    const sourceKeys = Object.keys(source);
+    for (const key of sourceKeys) {
+        if (destination[key] === undefined) {
+            destination[key] = source[key];
+        } else {
+            if (source[key] !== undefined && source[key] !== null) {
+                if (typeof destination[key] === 'object') {
+                    merge(destination[key], source[key]);
+                }
+            } else {
+                let a = 123;
```

Comment:
Looks like this else clause should be removed...

Reply:
indeed